chia/rpc/full_node_rpc_api.py

```diff
@@ -167,17 +168,49 @@ async def get_blockchain_state(self, _: Dict[str, Any]) -> EndpointResult:
         else:
             sync_progress_height = uint32(0)
 
+        average_block_time = uint32(0)
+        space = {"space": uint128(0)}
+
         if peak is not None and peak.height > 1:
             newer_block_hex = peak.header_hash.hex()
             # Average over the last day
-            header_hash = self.service.blockchain.height_to_hash(uint32(max(1, peak.height - 4608)))
-            assert header_hash is not None
-            older_block_hex = header_hash.hex()
+            older_header_hash = self.service.blockchain.height_to_hash(uint32(max(1, peak.height - 4608)))
+            assert older_header_hash is not None
+            older_block_hex = older_header_hash.hex()
             space = await self.get_network_space(
                 {"newer_block_header_hash": newer_block_hex, "older_block_header_hash": older_block_hex}
             )
-        else:
-            space = {"space": uint128(0)}
+
+            try:
+                newer_block: Optional[BlockRecord] = (
+                    await self.get_block_record({"header_hash": peak.header_hash.hex()})
+                )["block_record"]
+                while newer_block is not None and newer_block.height > 0 and not newer_block.is_transaction_block:
+                    newer_block = (await self.get_block_record({"header_hash": newer_block.prev_hash.hex()}))[
+                        "block_record"
+                    ]
+            except ValueError:
+                newer_block = None
+
+            if newer_block is not None:
+                older_header_hash = self.service.blockchain.height_to_hash(uint32(max(1, newer_block.height - 4608)))
+                try:
+                    older_block: Optional[BlockRecord] = (
+                        (await self.get_block_record({"header_hash": older_header_hash.hex()}))["block_record"]
```

Comment:
it seems unnecessary to use the JSON API to perform these lookups. You can just do:
```
older_block: Optional[BlockRecord] = self.service.blockchain.try_block_record(oler_header_hash)
if older_block is None:
    older_block = self.service.blockchain.get_block_record_from_db(older_header_hash)
```

Reply:
Fixed.